src/main/kotlin/com/hedvig/accountservice/schedulers/domain/MemberScheduler.kt

```diff
@@ -12,485 +54,509 @@ import org.axonframework.commandhandling.CommandHandler
-    }
-
-    @EventSourcingHandler
-    fun on(event: SchedulerDangerouslyResetEvent) {
-        currentState = event.nextState
-    }
+      )
+      logger.info { "Outside charge COMPLETED for memberId=${command.memberId} on transactionId=${command.transactionId}  and amount=${command.amount}" }
+    }
+  }
+
+  @EventSourcingHandler
+  fun on(event: ChargeCompletedEvent) {
+    currentState = event.nextState
+  }
+
+  @CommandHandler
+  fun handle(command: ChargeFailedCommand) {
+    if (command.transactionId == currentState.transactionId) {
+      val nextState = SchedulerState(
+        memberId = command.memberId,
+        status = ChargeStatus.CHARGE_FAILED,
+        changedBy = currentState.changedBy,
+        transactionId = command.transactionId
+      )
+      AggregateLifecycle.apply(ChargeFailedEvent(command.memberId, command.transactionId, command.failedAt, nextState))
+      logger.error { "Charge FAILED for memberId=${command.memberId} on transactionId=${command.transactionId}" }
+    } else {
+      logger.error { "Outside charge FAILED for memberId=${command.memberId} on transactionId=${command.transactionId}" }
+    }
+  }
+
+  @EventSourcingHandler
+  fun on(event: ChargeFailedEvent) {
+    currentState = event.nextState
+  }
+
+  @CommandHandler
+  fun handle(command: BlacklistSchedulerCommand): SchedulerStateTransition {
+    val nextState = SchedulerState(
+      memberId = command.memberId,
+      status = ChargeStatus.BLACKLISTED,
+      changedBy = command.blacklistedBy
+    )
+    val transition = isAllowedToTransition(nextState)
+    if (transition.outcome != SchedulerStateTransitionOutcome.ALLOWED_TRANSITION) {
+      apply(BlacklistSchedulerDeniedEvent(command.memberId, currentState, nextState))
+      logger.error { "Blacklist DENIED for memberId=${command.memberId} (Message: ${transition.message})" }
+      return transition
+    }
+    apply(SchedulerBlacklistedEvent(command.memberId, nextState))
+    logger.info { "Scheduler BLACKLISTED for memberId=${command.memberId}" }
+    return transition
+  }
+
+  @EventSourcingHandler
+  fun on(event: SchedulerBlacklistedEvent) {
+    currentState = event.nextState
+  }
+
+  @CommandHandler
+  fun handle(command: RevertBlacklistSchedulerCommand): SchedulerStateTransition {
+    val nextState = SchedulerState(
+      memberId = command.memberId,
+      status = ChargeStatus.INITIATED,
+      changedBy = command.revertedBy
+    )
+    val transition = isAllowedToTransition(nextState)
+    if (transition.outcome != SchedulerStateTransitionOutcome.ALLOWED_TRANSITION) {
+      apply(BlacklistSchedulerDeniedEvent(command.memberId, currentState, nextState))
+      logger.error { "Reverting Blacklist is DENIED for memberId=${command.memberId} (Message: ${transition.message})" }
+      return transition
+    }
+    apply(SchedulerInitiatedAfterBlacklistRevertedEvent(command.memberId, nextState))
```

Comment:
Not sure about the naming, since the event always contains a "nextState" property I think it's sufficient with 
```suggestion
    apply(SchedulerBlacklistRevertedEvent(command.memberId, nextState))
```